### PR TITLE
473 eg graphs randomize settings

### PR DESCRIPTION
**Pull request by egolol**
- [x] Add abstract function raddomize() to graph
- [x] Implement it efficiently (check with Mite) for the kinds of graphs we have
(any parameters should be in the settings for graph)
- [x] Document it

### CODE REVIEW

**Comment by annacanal** [comment]
I can continue tomorrow morning with the MultiplexGraphs.
@mitemijalkov @egolol 
Should we randomize A for each layer?
Or should we randomize the supra-adjacency matrix?

**Comment by egolol** [comment]
maybe its better if we have a skype meeting to discuss it, how about 11 sounds for you guys? @annacanal  @mitemijalkov

**Comment by annacanal** [comment]
> maybe its better if we have a skype meeting to discuss it, how about 11 sounds for you guys? @annacanal @mitemijalkov

11h works for me!

**Comment by giovannivolpe** [comment]
Also, add the respective tests (if needed, let's discuss it)
and double-check these functions with @mitemijalkov

**Comment by giovannivolpe** [comment]
Finally, the Dummy graph classes need a (simple) implementation of randomize (it can just do nothing and return the same graph), otherwise there are errors in the tests.

**Comment by annacanal** [comment]
> randomize

Done

**Comment by annacanal** [comment]
> Call the method randomize(g, varargin)

Done

**Comment by annacanal** [comment]
> Finally, the Dummy graph classes need a (simple) implementation of randomize (it can just do nothing and return the same graph), otherwise there are errors in the tests.

Done

**Comment by annacanal** [comment]
Graph settings were outdated. Now test_braph2 works.
Just missing: add the respective tests (if needed, let's discuss it). If needed, I suggest to leave it for another branch, so we can already merge this one!
@giovannivolpe @egolol @mitemijalkov

File: braph2/graph/graphs/DummyGraph.m [hunk 1]
```diff
@@ -126,4 +129,11 @@
             negativity_type =  Graph.NONNEGATIVE;
         end
     end
+    methods (Static)
+        function gr = randomize(g, varargin)
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyGraph.m:133]: Comment functions randomize

File: braph2/graph/graphs/GraphBD.m [hunk 2]
```diff
@@ -150,65 +162,76 @@
...
+        function [randomized_graph, swaps] = randomize(g, varargin)
+            % RANDOMIZE returns a randomized graph and the number of swaps.
+            %
+            % RANDOMIZED_GRAPH, SWAPS = RANDOMIZE(G) 
+            % returns the randomized graph RANDOMIZED_GRAPH obtained from a
+            % number of edge swaps SWAPS. Utilizes available graph
+            % settings.
+            %
+            % RANDOMIZED_GRAPH, SWAPS = RANDOMIZE(G, 'PROPERTYRULE1', 'VALUE1', ...) 
+            % returns the randomized graph RANDOMIZED_GRAPH obtained from a
```
Comment (giovannivolpe) [inline braph2/graph/graphs/GraphBD.m:175]: Add rule explicitly, since it's only one

File: braph2/graph/graphs/DummyGraph.m (hunk 1 above)
Comment (giovannivolpe) [reply braph2/graph/graphs/DummyGraph.m:133]: RANDOMIZED_GRAPH, SWAPS = RANDOMIZE(G, 'xxx', NUMBER_OF_SWAPS)

File: braph2/graph/graphs/GraphBD.m [hunk 3]
```diff
@@ -150,65 +162,77 @@
...
+    methods
+        function [randomized_graph, swaps] = randomize(g, varargin)
+            % RANDOMIZE returns a randomized graph and the number of swaps
+            %
+            % RANDOMIZED_GRAPH, SWAPS = RANDOMIZE(G) 
+            % returns the randomized graph RANDOMIZED_GRAPH obtained from a
+            % number of edge swaps SWAPS. Utilizes available graph
+            % settings.
+            %
+            % RANDOMIZED_GRAPH, SWAPS = RANDOMIZE(G, 'ATTEMPTSPEREDGE', 'NUMERICVALUE')
```
Comment (giovannivolpe) [inline braph2/graph/graphs/GraphBD.m:174]: [RANDOMIZED_GRAPH, SWAPS] = RANDOMIZE(G, 'GraphBD.AttemptsPerEdge', NUMBER)

Comment (giovannivolpe) [reply braph2/graph/graphs/GraphBD.m:174]: make the same change also to the other Graphs.

**Pull request merged**